static/inline/js/libs/player.js

```diff
@@ -0,0 +1,304 @@
+        this.playIconTemplate = isset(options.playIconTemplate) ? options.playIconTemplate : null;
+        this.onResize = isset(options.onResize) && isFunction(options.onResize) ? options.onResize : null;
+        this.onNext = isset(options.onNext) && isFunction(options.onNext) ? options.onNext : null;
+
+        this.instance = null;
+        this.el = 'string' === typeof el ? document.getElementById(el) : el;
+
+        this.assets = (function(data){
+            var assets = [];
+            for (var i = 0; i < data.length; i++) {
+                var video = data[i];
+
+                if (isEmpty(video)) break;
+
+                var asset = video.asset;
+                asset.assetId = video.id;
+                asset.assetTitle = video.title;
+                asset.loginId = video.loginId;
+
+                if (isset(video,'events') && video.events.length) {
+                    asset.analyticsLogUrl = video.analytics;
+                    asset.analyticsObj = video.events[1].data;
+                } else {
+                    asset.analyticsObj = {
+                        pg: isset(video,'parentId') ? video.parentId : ( isset(options,'channel') ? options.channel.id : 0 ),
+                        vd: video.id,
+                        li: video.loginId
+                    };
+                }
+
+                if (!asset.sources) asset.sources = [{ file: asset.videoId }];
+                asset.type = asset.type || 'youtube';
+                assets.push(asset);
+            }
+            return assets;
+        }(options.data));
+
+
+        //Context reference for Methods.
+        var that = this;
+
+        //Sometimes we want/need to show an intearctive overlay on top of the player. We need this for MP4 videos that will
+        //cue (mobile or autoplay:off) to actual play the video on demand.
+        this.addOverlay = function(asset){
+            var overlay = this.el.querySelector('.tvp-overlay');
+
+            if (!Boolean(overlay)){
+                overlay = document.createElement('div');
+                overlay.classList.add('tvp-overlay');
+            }            
+            
+            overlay.style.backgroundImage = 'url("' + asset.thumbnailUrl + '")';
+
+            var overlayTemplate = Utils.tmpl(that.playerOverlayTemplate, {
+                thumbnailUrl: asset.thumbnailUrl,
+                overlayOpacity: this.overlayOpacity,
+                overlayColor: this.overlayColor ? '#' + this.overlayColor : 'transparent',
+                playButtonHeight: this.playButtonHeight
+            });
+
+            overlay.innerHTML = overlayTemplate;
+
+            this.el.appendChild(overlay);
+            var playerHolder = this.el.querySelector('.tvp-play-holder');
+            playerHolder.innerHTML = this.playIconTemplate;
+            // playerHolder.querySelector('.tvp-video-play').style.backgroundColor = this.playButtonBackgroundColor
```

Comment:
this is used for having a play icon for mp4 type videos. since we're applying the latest version (3.0.1) of the player lib, this will be remove